### PR TITLE
Eq 410 composite question types

### PR DESCRIPTION
**Pull request by weapdiv-david**
### What is the context of this PR?

This PR builds upon PR #318 

This PR replaces all questions in all schemas with either a "General" or "DateRange" question type.  Two question subclasses are created to represent them.  The parser is also updated to throw an exception when encountering a question type that is NOT one of these types. It also subclasses the answer types currently supported.  Existing templates are not not affected.

This PR lays one of the foundation stones for the larger display-widget refactoring that's currently taking place.
### How to review

1) Check out the branch
2) Run the unit tests
3) Run the app and ensure the questionnaires are rendered as before
- [ ] Do all commits have sensible commit messages?
- [ ] Is all new code unit tested?
- [ ] Are there integration tests if needed?
- [ ] Is there sufficient logging?
- [ ] Is there documentation or is the code self-describing?
### Who worked on the PR

Describe who worked on the changes, so that other people can review.

### CODE REVIEW

File: app/schema/answer.py [hunk 1]
```diff
@@ -15,3 +14,45 @@ def __init__(self):
         self.templatable_properties = []
         self.options = []
         self.alias = None
+
+    @staticmethod
+    def get_instance(answer_type):
+        # Do the imports here to avoid circular dependencies
+        # TODO: Refactor out into a factory of some sort
```
Comment (warrenbailey) [inline app/schema/answer.py]: Maybe add the factory now, this looks pretty messy.

File: app/schema/answer.py [hunk 2]
```diff
@@ -15,3 +14,45 @@ def __init__(self):
...
+        from app.schema.answers.integer_answer import IntegerAnswer
+        from app.schema.answers.percentage_answer import PercentageAnswer
+        from app.schema.answers.positiveinteger_answer import PositiveIntegerAnswer
+        from app.schema.answers.radio_answer import RadioAnswer
+        from app.schema.answers.textarea_answer import TextareaAnswer
+        from app.schema.answers.textfield_answer import TextfieldAnswer
+
+        answer_type = answer_type.upper()
+
+        if answer_type == 'CHECKBOX':
```
Comment (warrenbailey) [inline app/schema/answer.py]: If we implemented the factory we could avoid all these elif block by implementing a register/create pattern

For example:

```
class AnswerFactory(object):
    def __init(self):
         self.classes = {}
    def register(self, name, class):
         self.classes[name] = class

    def create(name):
        return self.classes[name]()
```

Then register all the types like this:

```
answer_factory.register("CURRENCY", CurrencyAnswer)
```

File: app/schema/answer_factory.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+from app.schema.answers.integer_answer import IntegerAnswer
+from app.schema.answers.percentage_answer import PercentageAnswer
+from app.schema.answers.positiveinteger_answer import PositiveIntegerAnswer
+from app.schema.answers.radio_answer import RadioAnswer
+from app.schema.answers.textarea_answer import TextareaAnswer
+from app.schema.answers.textfield_answer import TextfieldAnswer
+
+
+class AnswerFactory(object):
+    @staticmethod
```
Comment (collisdigital) [inline app/schema/answer_factory.py]: There's a generic factory in (https://github.com/ONSdigital/eq-survey-runner/blob/master/app/utilities/factory.py) that does this, you can just create one of them here and register the types e.g

AnswerFactory = factory.Factory()
AnswerFactory.register_all(...)

File: app/schema/question_factory.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from app.schema.questionnaire import QuestionnaireException
+from app.schema.questions.general_question import GeneralQuestion
+from app.schema.questions.date_range_question import DateRangeQuestion
+
+
+class QuestionFactory(object):
```
Comment (collisdigital) [inline app/schema/question_factory.py]: See other comment about https://github.com/ONSdigital/eq-survey-runner/blob/master/app/utilities/factory.py

**Comment by warrenbailey** [comment]
LGTM 👍

**Pull request merged**